### ISSUE TITLE
Fullscreen container dimensions change when opening windows behind it

### ISSUE THREAD

**Issue by Xyene** [swaywm/sway#6225]
This has implications for cursor images.

This behavior is easy to observe after applying https://github.com/swaywm/sway/pull/6224, but https://github.com/swaywm/sway/pull/6224 does not introduce the issue.

To reproduce:

1. open `V[kitty]`
2. observe cursor changes to `vertical_text`
3. `fullscreen`
4. use keybinding to open another `kitty` underneath the fullscreen view
5. depending on whether the cursor was on the left half or right half of the screen when performing 5., the cursor has either remained the same, or switched to `left_ptr`

The expected result is that the cursor remains `vertical_text`.

I think I've ~tracked down how this happens. https://github.com/swaywm/sway/blob/4e77bc293515ea2f00ebbc4395cb0eb4446d0195/sway/input/cursor.c#L209

is called with `node = NULL`, which resets the cursor to `left_ptr`.

https://github.com/swaywm/sway/blob/4e77bc293515ea2f00ebbc4395cb0eb4446d0195/sway/input/cursor.c#L225-L227

This is triggered by a call to `handle_rebase` upon container open, where the result of `node_at_coords` is `NULL` (it shouldn't be -- it should be the fullscreen view).

https://github.com/swaywm/sway/blob/8529141150ef2d9a07b9c0fad193bdbc12d3ad42/sway/input/seatop_default.c#L754-L760
https://github.com/swaywm/sway/blob/8529141150ef2d9a07b9c0fad193bdbc12d3ad42/sway/input/seatop_default.c#L767-L770

Digging into `node_at_coords`, we hit:

https://github.com/swaywm/sway/blob/4e77bc293515ea2f00ebbc4395cb0eb4446d0195/sway/input/cursor.c#L146-L151

and return `NULL`. `tiling_container_at` returns `NULL` after returning the value from `view_container_at`:

https://github.com/swaywm/sway/blob/eb9e77f4eae292da483191dcac18dbdfa50b984e/sway/tree/container.c#L360-L365

Finally, printing the `wlr_box` of the fullscreen view inside `view_container_at` demonstrates what is happening.

https://github.com/swaywm/sway/blob/eb9e77f4eae292da483191dcac18dbdfa50b984e/sway/tree/container.c#L345-L351

When first fullscreening the window, we see:

```
[sway/tree/container.c:352] (lx = 1955.564737, ly = 974.010224) in [x = 1440, y = 530] x [w = 2560, h = 1440]
```

Once opening the `kitty` under the fullscreen view, upon the `handle_rebase` call stack and without having moved the cursor:

```
[sway/tree/container.c:352] (lx = 1955.564737, ly = 974.010224) in [x = 1440, y = 550] x [w = 1275, h = 1420]
```

In other words, `container->pending.width` and `height` changed as if the fullscreen view was still tiled, and are used for fullscreen hittesting.

/cc @RPigott -- thoughts?

**Comment by purxiz** [comment]
I can confirm this issue, it's particularly irritating when fullscreening a video in web applications, since it seems as though they create a new fullscreen window to accomplish that, which leads to all the windows underneath being resized and re-containered.
